RestaurantMap/Scener/Restaurant/RestaurantViewController.swift

```diff
@@ -9,11 +9,28 @@ import UIKit
 
 class RestaurantViewController: UIViewController {
     
+    @IBOutlet private weak var nameLabel: UILabel!
+    @IBOutlet private weak var rateLabel: UILabel!
+    @IBOutlet private weak var reviewCountLabel: UILabel!
+    @IBOutlet private weak var priceLabel: UILabel!
+    @IBOutlet private weak var addressLabel: UILabel!
+    @IBOutlet private weak var imageView: UIImageView!
+    
     private var router: RestaurantRoutingLogics?
     private var interactor: RestaurantBusinessLogics?
+    
+    var restaurant: Restaurant?
+    
+    static func restaurantViewController(restaurant: Restaurant) -> RestaurantViewController {
```

Comment:
Why need to create this static function? Attend that using too many static variables / functions will lead to leak memory. I think you shouldn't take this habit.

Reply:
In HomeRouter, you can write like this

let vc = RestaurantViewController()
vc.restaurant = restaurant
navigationController?.present(restaurantViewController, animated: true)

This is okay, no need to create a static function

Reply:
I using static func instead of assigning value for internal variables in another class, I    ###think calling static func to assign a value to another class won't affect the memory. I tested your way and mine on Xcode and as a result, my method takes less memory space.
![Screen Shot 2022-04-18 at 10 03 13](https://user-images.githubusercontent.com/102586283/163749630-dd950b77-2137-4326-90d3-548fb2775594.png)
![Screen Shot 2022-04-18 at 10 06 05](https://user-images.githubusercontent.com/102586283/163749638-d3c43466-f76f-4b9e-9e88-b2a450012bf5.png)
 You can refer to the documentation:  https://developer.apple.com/swift/blog/?id=7#:~:text=The%20lazy%20initializer,mark%20it%20private.
 Here is what I can understand.
 Thanks for the perfect comment  <3 and looking forward to hearing from you